### ISSUE TITLE
RCS1175 must not appear when first parameter of extension method is not used

### ISSUE THREAD

**Issue by xperiandri** [JosefPihrt/Roslynator#117]
Look at this code excerpt.
Even though `this ControllerBase controller` is not used RCS1175 must not appear as first parameter is used as extension method binder.

``` C#
public class UnprocessableEntityObjectResult : ObjectResult
{
    public UnprocessableEntityObjectResult(ModelStateDictionary modelState) : base(new SerializableError(modelState))
    {
        StatusCode = StatusCodes.Status422UnprocessableEntity;
    }
}

public static class StatusCodeControllerExtensions
{
    public static UnprocessableEntityObjectResult UnprocessableEntity(this ControllerBase controller, ModelStateDictionary modelState)
        => new UnprocessableEntityObjectResult(modelState);
}
```

**Comment by JosefPihrt** [comment]
This issue has been discussed here #55.

Originally, all unused parameter were reported by RCS1163. Then it turned out that 'this' parameters are somewhat special so they were separated into new analyzer RCS1175. This separation enables you to completely disable analysis of unused 'this' parameters if that is your preference. Other possible solution is to suppress reported diagnostic (using pragma or SuppressMessageAttribute).

**Comment by xperiandri** [comment]
But it displays me RCS1175 in fix UI

**Comment by JosefPihrt** [comment]
I'm sorry but I don't understand what do you mean by that.

**Comment by xperiandri** [comment]
![image](https://user-images.githubusercontent.com/2365592/28242714-c54c90f4-69bb-11e7-9b02-a45808242212.png)

**Comment by JosefPihrt** [comment]
How did you suppress the diagnostic? I see neither pragma directive nor SuppressMessageAttribute in your code.

**Comment by xperiandri** [comment]
I didn't.
You told me that you excluded this case from RCS1175 to RCS1163. This mean that RCS1163 must appear in my case, mustn't it?

**Comment by JosefPihrt** [comment]
I separated unused 'this' parameters **from** RCS1163 **to** RCS1175. Not the other way around.

Unused parameters except 'this' parameters are reported by RCS1163.
Unused 'this' parameters are reported by RCS1175.

**Comment by xperiandri** [comment]
Ah, I misunderstood you.
I thought vice versa.
